consensus/polybft/state_transaction.go

```diff
@@ -4,108 +4,67 @@ import (
 	"bytes"
 	"errors"
 	"fmt"
-	"math/big"
-	"reflect"
 
 	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/contracts"
 	"github.com/0xPolygon/polygon-edge/crypto"
 	"github.com/0xPolygon/polygon-edge/state/runtime/precompiled"
 	"github.com/0xPolygon/polygon-edge/types"
-	"github.com/mitchellh/mapstructure"
-	"github.com/umbracle/ethgo/abi"
 )
 
-var (
-	commitmentABIType = abi.MustNewType("tuple(uint256 startId, uint256 endId, bytes32 root)")
-
-	commitABIMethod = contractsapi.StateReceiver.Abi.Methods["commit"]
-)
-
-// StateTransactionInput is an abstraction for different state transaction inputs
-type StateTransactionInput interface {
-	// EncodeAbi contains logic for encoding arbitrary data into ABI format
-	EncodeAbi() ([]byte, error)
-	// DecodeAbi contains logic for decoding given ABI data
-	DecodeAbi(b []byte) error
-	// Type returns type of state transaction input
-	Type() StateTransactionType
-}
-
-type StateTransactionType string
-
 const (
 	abiMethodIDLength      = 4
 	stTypeBridgeCommitment = "commitment"
 	stTypeEndEpoch         = "end-epoch"
 )
 
-// Commitment holds merkle trie of bridge transactions accompanied by epoch number
-type Commitment struct {
+// PendingCommitment holds merkle trie of bridge transactions accompanied by epoch number
+type PendingCommitment struct {
+	*contractsapi.StateSyncCommitment
 	MerkleTree *MerkleTree
 	Epoch      uint64
-	FromIndex  uint64
-	ToIndex    uint64
 }
 
 // NewCommitment creates a new commitment object
-func NewCommitment(epoch uint64, stateSyncEvents []*types.StateSyncEvent) (*Commitment, error) {
+func NewCommitment(epoch uint64, stateSyncEvents []*contractsapi.StateSyncedEvent) (*PendingCommitment, error) {
```

Comment:
Does it make sense to rename this to `NewPendingCommitment`?

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1166/commits/0761e2dc8fcade2f8f976bb1976629310e14bacf